src/io/flutter/utils/FlutterModuleUtils.java

```diff
@@ -80,43 +78,52 @@ public static boolean usesFlutter(@NotNull Project project) {
   }
 
   /**
-   * Create a Flutter run configuration.
-   *
-   * @param project the project
-   * @param main    the target main
+   * Creates a Flutter run configuration if none exist.
    */
-  public static void createRunConfig(@NotNull Project project, @Nullable VirtualFile main) {
-    final ConfigurationFactory[] factories = FlutterRunConfigurationType.getInstance().getConfigurationFactories();
-    final Optional<ConfigurationFactory> factory =
-      Arrays.stream(factories).filter((f) -> f instanceof FlutterRunConfigurationType.FlutterConfigurationFactory).findFirst();
-    assert (factory.isPresent());
-    final ConfigurationFactory configurationFactory = factory.get();
+  public static void autoCreateRunConfig(@NotNull Project project, @NotNull PubRoot root) {
+    final VirtualFile main = root.getLibMain();
+    if (main == null || !main.exists()) return;
+
+    final FlutterRunConfigurationType configType = FlutterRunConfigurationType.getInstance();
 
     final RunManager runManager = RunManager.getInstance(project);
-    final List<RunConfiguration> configurations = runManager.getConfigurationsList(FlutterRunConfigurationType.getInstance());
-
-    // If the target project has no flutter run configurations, create one.
-    if (configurations.isEmpty()) {
-      final RunnerAndConfigurationSettings settings =
-        runManager.createRunConfiguration(project.getName(), configurationFactory);
-      final SdkRunConfig config = (SdkRunConfig)settings.getConfiguration();
-
-      // Set config name.
-      final String name = config.suggestedName();
-      if (name == null) {
-        config.setName(project.getName());
-      }
+    if (!runManager.getConfigurationsList(configType).isEmpty()) {
+      return;
+    }
 
-      // Set fields.
-      final SdkFields fields = new SdkFields();
-      if (main != null && main.exists()) {
-        fields.setFilePath(main.getPath());
-      }
-      config.setFields(fields);
+    final RunnerAndConfigurationSettings settings =
+      runManager.createRunConfiguration(project.getName(), configType.getFactory());
+
+    final SdkRunConfig config = (SdkRunConfig)settings.getConfiguration();
 
-      runManager.addConfiguration(settings, false);
-      runManager.setSelectedConfiguration(settings);
+    // Set config name.
+    final String name = config.suggestedName();
+    if (name == null) {
+      config.setName(project.getName());
```

Comment:
Reproduced.

It looks like the templates for "flutter create" use "main.dart" as the name, and it's not detected here because .idea/runConfigurations isn't aren't loaded from disk yet.

I removed the calls to create the run configuration in FlutterModuleBuilder because the template already does it. Also, when opening a project, I moved the run configuration creation logic to happen later. It looks like the run configurations are loaded by then.

Also changed the name to "main.dart" to match the template.